src/lumigo_tracer/spans_container.py

```diff
@@ -150,44 +150,47 @@ def add_span(self, span: dict) -> dict:
         This function parses an request event and add it to the span.
         """
         new_span = recursive_json_join(span, self.base_msg)
-        self.spans.append(new_span)
-        self.span_ids_to_send.add(span["id"])
+        span_id = new_span["id"]
+        self.spans[span_id] = new_span
+        self.span_ids_to_send.add(span_id)
         return new_span
 
-    def get_last_span(self) -> Optional[dict]:
-        if not self.spans:
+    def get_span_by_id(self, span_id: Optional[str]) -> Optional[dict]:
+        if not span_id:
             return None
-        return self.spans[-1]
+        return self.spans.get(span_id)
 
-    def get_span_by_id(self, span_id: str) -> Optional[dict]:
-        for span in self.spans:
-            if span.get("id") == span_id:
-                return span
-        return None
-
-    def pop_last_span(self) -> Optional[dict]:
-        return self.spans.pop() if self.spans else None
+    def pop_span(self, span_id: Optional[str]) -> Optional[dict]:
+        if not span_id:
+            return None
```

Comment:
I don't think that we need those lines, you use `dict.pop` with default value

Reply:
mypy otherwise fail